sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -1131,17 +1130,24 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog with QueryErrorsB
    * remove extra project which only re-assign expr ids from the plan so that we can identify exact
    * duplicates metric definition.
    */
-  private def simplifyPlanForCollectedMetrics(plan: LogicalPlan): LogicalPlan = {
+  def simplifyPlanForCollectedMetrics(plan: LogicalPlan): LogicalPlan = {
     plan.resolveOperators {
       case p: Project if p.projectList.size == p.child.output.size =>
-        val assignExprIdOnly = p.projectList.zipWithIndex.forall {
+        var assignExprIdOnly = p.projectList.zipWithIndex.forall {
           case (Alias(attr: AttributeReference, _), index) =>
             // The input plan of this method is already canonicalized. The attribute id becomes the
             // ordinal of this attribute in the child outputs. So an alias-only Project means the
             // the id of the aliased attribute is the same as its index in the project list.
             attr.exprId.id == index
           case _ => false
```

Comment:
I think we just need to add one more case match here
```
// Both Attribute has been canonicalized to use index as the attr id, so `semanticEquals` still works here.
case (left: Attribute, right: Attribute) => left.semanticEquals(right)
```

Reply:
Actual it is not. Because `Alias` is not `Attribute`, we will miss the `Alias` case match.

Reply:
It is Surprising for me to find that `Alias` is an expression.

Reply:
I'm totally confused
```
case (Alias(attr: AttributeReference, _), index) => ...
case (attr: AttributeReference, index) => ...
```
Why wouldn't the above work?

Reply:
Interesting, the above didn't work I believe. Let me take a closer look by debugger.

Reply:
I am not sure how the existing behaves, but in my example I check the case where the expr.id is 1 but the index is 0.

<img width="762" alt="Screenshot 2023-09-18 at 2 19 49 PM" src="https://github.com/apache/spark/assets/1938382/e6850683-9557-4382-b203-be124c50606a">